### DISCUSSION TITLE
Pagination doesn't work

### DISCUSSION THREAD

**Discussion by BolotnikovMS** [adonisjs/core#4363] [Help]
Hi all. When I try to do pagination, nothing happens Adonis 6 ((( Example code:
const posts = await Post.query()
      .if(page && limit, (query) => query.paginate(page, limit))

**Comment by tomgobich** [comment]
Hey there!

Are you certain you have both a `page` and `limit` value that results in a truthy expression? Pagination has been working just fine for me so far in version 6.

Also - you can provide formatted code examples using Markdown's three backticks and a lang code (```ts), which will look like the below:
```ts
const page = 1
const limit = 10
const posts = await Post.query().paginate(page, limit)
```

**Reply by BolotnikovMS** [reply]
Hello, sorry for such a long response. Yes, the values are transferred. But the point is that I am writing an API and I need to check for the presence of pages in the limit and if they exist, then break them into pages, otherwise give away all the records. But when using the check and pass page, nothing happens to the limit.
```
const { page, limit } = request.qs()
const posts = await Posts.query()
      .if(page && limit, (query) => query.paginate(page, limit))
```

**Comment by holoyan** [comment]
You should have some validation for the `limit`, otherwise anyone can write larg number and that is not good, imagine loading 1,000,000 posts at once, 
so you can do something like this and get rid of `if` condition

```typescript
  const limit = Math.min(100, request.input('limit', 10)); // if no limit is passed then default will be 10, if limit is bigger then 100 then Math.min will return 100, so you restrcit max value by 100;

  const page = request.input('page', 1); // same here

const posts = await Post.query().paginate(page, limit)

```

**Reply by BolotnikovMS** [reply]
Thanks for the answer)
